elastic/kibana
# [Reporting] Update Puppeteer to version 8.0.0 and Chromium to r856583

Implements #90496

Closes https://github.com/elastic/kibana/issues/89555
Closes https://github.com/elastic/kibana/issues/89467
Closes https://github.com/elastic/kibana/issues/87956

**Release note:** Updated the headless browser for Kibana Reporting to Chromium r856583 (90.0.4427.0)

 - Updates Puppeteer to version 8.0.0 and Chromium to r856583
 - Links to new build of Linux headless_shell in the Kibana team GCS bucket
 - Links to main download site of Chromium for Mac and Windows
 - Removes Mac and Windows compatibility from the Chromium build scripts

----
Version information for r856583:
  https://storage.googleapis.com/chromium-find-releases-static/d16.html#d163fd7c6905e456501aba914a24c20b62b4c6dc

Explanation of headless_shell: 
  https://chromium.googlesource.com/chromium/src/+/d163fd7c6905e456501aba914a24c20b62b4c6dc/headless/#usage-as-a-c_library

----
**How to test this PR**
These steps need to happen for every platform: Mac, Windows, Linux x64 and Linux arm64
 - Make sure the Reporting plugin is fetching the correct version of the browser at start-up time, and that it can successfully unzip it and copy the files to `x-pack/plugins/reporting/chromium`
 - Make sure there are no errors when using the [Reporting diagnostic tool](https://www.elastic.co/guide/en/kibana/current/reporting-troubleshooting.html#reporting-diagnostics)
 - All functional and API tests that generate PDF and PNG files should pass
 - Search for [issues in the Puppeteer repo](https://github.com/puppeteer/puppeteer/issues) that might apply to our usage

----
**Metrics**
Build size changes, before and after:
| Package                                    | Before | After |
| ------------------------------------------ | ---- | ---- |
| kibana-8.0.0-SNAPSHOT-linux-x86_64.tar.gz  | 285M | 289M |
| kibana-8.0.0-SNAPSHOT-amd64.deb            | 289M | 293M |
| kibana-8.0.0-SNAPSHOT-x86_64.rpm           | 293M | 296M |
| kibana-8.0.0-SNAPSHOT-linux-aarch64.tar.gz | 297M | 301M |
| kibana-8.0.0-SNAPSHOT-arm64.deb            | 301M | 306M |
| kibana-8.0.0-SNAPSHOT-aarch64.rpm          | 304M | 309M |
| kibana-8.0.0-SNAPSHOT-docker-image.tar.gz  | 385M | 389M |
| kibana-8.0.0-SNAPSHOT-darwin-x86_64.tar.gz | 293M | **325M** |
| kibana-8.0.0-SNAPSHOT-windows-x86_64.zip   | 308M | **405M** |

----
**Checklist**
- [x] Functional test for "large" dashboard (over 11MP)
- [x] Run tests for Mac
- [x] Run tests for Windows
- [x] Run tests for Linux x64
- [x] Run tests for Linux arm64
  - Started an EC2 instance using Ubuntu 20.10 ARM, and installed the OS dependencies: `sudo apt install -y libnss3 fonts-liberation libfontconfig1`
